springCRUD/src/main/java/com/cnu2016/controller/OrdersController.java

```diff
@@ -20,67 +25,142 @@
         orders.setStatus("In Cart");
-        return ResponseEntity.status(HttpStatus.OK).body(ordersRepository.save(orders));
+        orders = ordersRepository.save(orders);
+        return ResponseEntity.status(HttpStatus.CREATED).body(new OrderSerializer(orders.getOrderID()));
+    }
+    @RequestMapping(value = "/api/orders/{oid}", method = RequestMethod.GET)
+    public ResponseEntity<?> getOrder(@PathVariable Integer oid) {
+        Orders order = ordersRepository.findOne(oid);
+        if (order == null || order.getStatus().equals("Deleted")) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid order id");
+        }
+        return ResponseEntity.status(HttpStatus.OK).body(new OrderSerializer(order.getOrderID()));
     }
-
     /**
      * Adds to an existing order.
-     * Will return 404 if productID or quantity is missing.
+     * Will return NOT_FOUND if any of the following is true
+     *      1) Order is missing
+     *      2) ProductID is missing
+     *      3) Quantity is missing
+     * Will return BAD_REQUEST if status is not "In Cart"
      * Will fetch price from product table
      */
-    @RequestMapping(value = "/api/order/{orderID}/orderLineItem", method = RequestMethod.POST)
+    @RequestMapping(value = "/api/orders/{orderID}/orderLineItem", method = RequestMethod.POST)
     public ResponseEntity<?> addToOrder(@RequestBody OrderDetailsSerializer orderDetailsSerializer, @PathVariable Integer orderID) {
-        if (orderDetailsSerializer == null || orderDetailsSerializer.getQuantity() == null || orderDetailsSerializer.getProductID() == null) {
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid data");
-        }
         Orders order = ordersRepository.findOne(orderID);
-        Product product = productRepository.findOne(orderDetailsSerializer.getProductID());
-        if (order == null || product == null) {
+        if (order == null || order.getStatus().equals("Deleted") == true) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid order id");
+        }
+        if (orderDetailsSerializer == null || orderDetailsSerializer.getQty() == null || orderDetailsSerializer.getProduct_id() == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Invalid data");
+        }
+        Product product = productRepository.findOne(orderDetailsSerializer.getProduct_id());
+        if (product == null) {
             return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid data");
         }
-        if (order.getStatus().equals("In Cart") == false || product.getQuantityInStock() < orderDetailsSerializer.getQuantity()) {
+        if (order.getStatus().equals("In Cart") == false) {
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Closed order or insufficient stock");
         }
+
+        //quick fix: must make it much faster
+        Integer productID = orderDetailsSerializer.getProduct_id();
```

Comment:
Use camelCase getProductId(). See below for review comments. 
https://docs.google.com/spreadsheets/d/1hqu6WtRwc5FxdsyWuhOV1RY0LfSTTLluMXByBHWqCBY/edit#gid=56856103